tesseract/tesseract_planning/tesseract_process_managers/src/process_planning_server.cpp

```diff
@@ -0,0 +1,429 @@
+  registerProcessPlanner(process_planner_names::RASTER_G_FT_PLANNER_NAME, createRasterGlobalGenerator());
+  registerProcessPlanner(process_planner_names::RASTER_FT_DT_PLANNER_NAME, createRasterDTGenerator());
+  registerProcessPlanner(process_planner_names::RASTER_FT_WAAD_PLANNER_NAME, createRasterWAADGenerator());
+  registerProcessPlanner(process_planner_names::RASTER_FT_WAAD_DT_PLANNER_NAME, createRasterWAADDTGenerator());
+  registerProcessPlanner(process_planner_names::RASTER_O_G_FT_PLANNER_NAME, createRasterOnlyGlobalGenerator());
+  registerProcessPlanner(process_planner_names::RASTER_CT_PLANNER_NAME, createRasterCTGenerator());
+  registerProcessPlanner(process_planner_names::RASTER_O_CT_PLANNER_NAME, createRasterOnlyCTGenerator());
+  registerProcessPlanner(process_planner_names::RASTER_CT_DT_PLANNER_NAME, createRasterCTDTGenerator());
+  registerProcessPlanner(process_planner_names::RASTER_CT_WAAD_PLANNER_NAME, createRasterCTWAADGenerator());
+  registerProcessPlanner(process_planner_names::RASTER_CT_WAAD_DT_PLANNER_NAME, createRasterCTWAADDTGenerator());
+  registerProcessPlanner(process_planner_names::RASTER_G_CT_PLANNER_NAME, createRasterGlobalCTGenerator());
+  registerProcessPlanner(process_planner_names::RASTER_O_G_CT_PLANNER_NAME, createRasterOnlyGlobalCTGenerator());
+}
+
+bool ProcessPlanningServer::hasProcessPlanner(const std::string& name) const
+{
+  return (process_planners_.find(name) != process_planners_.end());
+}
+
+std::vector<std::string> ProcessPlanningServer::getAvailableProcessPlanners() const
+{
+  std::vector<std::string> planners;
+  planners.reserve(process_planners_.size());
+  for (const auto& planner : process_planners_)
+    planners.push_back(planner.first);
+
+  return planners;
+}
+
+ProcessPlanningFuture ProcessPlanningServer::run(const ProcessPlanningRequest& request)
+{
+  CONSOLE_BRIDGE_logInform("Tesseract Planning Server Recieved Request!");
+  ProcessPlanningFuture response;
+  response.plan_profile_remapping = std::make_unique<const PlannerProfileRemapping>(request.plan_profile_remapping);
+  response.composite_profile_remapping =
+      std::make_unique<const PlannerProfileRemapping>(request.composite_profile_remapping);
+
+  response.input = std::make_unique<Instruction>(request.instructions);
+  const auto* composite_program = response.input->cast_const<CompositeInstruction>();
+  ManipulatorInfo mi = composite_program->getManipulatorInfo();
+  response.global_manip_info = std::make_unique<const ManipulatorInfo>(mi);
+
+  if (!isNullInstruction(request.seed))
+    response.results = std::make_unique<Instruction>(request.seed);
+  else
+    response.results = std::make_unique<Instruction>(generateSkeletonSeed(*composite_program));
+
+  auto it = process_planners_.find(request.name);
+  if (it == process_planners_.end())
+  {
+    CONSOLE_BRIDGE_logError("Requested motion planner is not supported!");
```

Comment:
Should be "Requested Taskflow is not supported"? I suppose the users don't need to know the details that it is a taskflow, but maybe something that has the sense that it involves multiple processes. Maybe Process pipeline?

Reply:
Updated.